### PR TITLE
imputed_birthday takes YYYY-MM rather than just YYYY

### PR DESCRIPTION
**Pull request by seb231**
Function takes birth month (2003-01) and report date

earliest date is always first of the birth month

latest date is either then end of the month or if the report date is the same month as the birth month, then `report date - 1` is the latest date

As before then a random date is picked between those two dates

### CODE REVIEW

File: src/eda.R [hunk 1]
```diff
@@ -311,7 +319,11 @@ placement.transitions <- episodes %>% group_by(period_id, phase_number, placemen
   filter(!is.na(next_placement)) %>%
   mutate(transition_year = year(end + days(275)))
 
-placement.transitions.grouped <- placement.transitions %>% mutate(admission_age = transition_year - DOB) %>%
+transition_age <- function(end_of_placment, month_of_birth) {
+  floor(time_length(difftime(end_of_placment, as.Date(paste0(month_of_birth, "-01"), "%Y-%m-%d")), "years"))
+}
+
+placement.transitions.grouped <- placement.transitions %>% mutate(admission_age = transition_age(end,DOB)) %>%
```
Comment (henrygarner) [inline src/eda.R]: placement.transitions could be updated to include the birthday we have already imputed for the child?

File: src/eda.R [hunk 2]
```diff
@@ -707,7 +720,7 @@ for (age.entry in 0:17) {
                                      .range(["#4E79A7","#A0CBE8","#F28E2B","#FFBE7D","#59A14F","#8CD17D","#B6992D","#F1CE63","#499894","#86BCB6",
                                      "#E15759","#FF9D9A","#79706E","#BAB0AC","#D37295","#FABFD2","#B07AA1","#D4A6C8","#9D7660","#D7B5A6", "#FFFFFF", "#FFFFFF"])')) %>%
       htmlwidgets::prependContent(htmltools::tags$h2(paste("Age", age.entry, "joiners in", year.entry))) %>%
-      widget2png(paste0("age_", age.entry, "_year_", year.entry, "_incl_open.png"))
+      widget2png(chart_path(paste0("age_", age.entry, "_year_", year.entry, "_incl_open.png")))
```
Comment (henrygarner) [inline src/eda.R:713]: :+1:

File: src/eda.R [hunk 3]
```diff
@@ -976,7 +1008,7 @@ ggsave(chart_path("joiners-leaver-rate.png"), width = 11, height = 8)
 
 monthly_leavers_age <- periods %>%
   mutate(month = floor_date(end, "month"),
-         exit_age = round(as.numeric(end - as.Date(paste0(birth_year,"-07-31"))) / 365.0)) %>%
+         exit_age = round(as.numeric(end - birthday) / 365.0)) %>%
```
Comment (henrygarner) [inline src/eda.R:1001]: 👍

File: src/eda.R [hunk 4]
```diff
@@ -114,6 +109,21 @@ birthday_before_date <- function(birth_date, other_date) {
   }
 }
 
+
+imputed_birthday <- function(birth_year_month, report_date) {
```
Comment (henrygarner) [inline src/eda.R]: This function is already updated in master. The implementation there also takes account of the episodes data to protect against children being negative ages at join or over 18 when they leave, so I think it's preferable.

**Review by henrygarner — CHANGES_REQUESTED** [review]
We've definitely overlapped on the imputed_birthday function, but there's an opportunity to make the placement.transitions.grouped data consistent here, if you up for making a small change.

File: src/eda.R (hunk 4 above)
Comment (seb231) [reply src/eda.R]: I thought I had covered that, but I'll take your word for it and switch it back to your implementation 👍

**Review by henrygarner — APPROVED** [review]
LGTM

**Pull request merged**